relay-server/src/envelope.rs

```diff
@@ -710,20 +710,31 @@ impl Item {
     }
 
     /// Sets the payload and content-type of this envelope.
-    pub fn set_payload<B>(&mut self, content_type: ContentType, payload: B)
-    where
+    pub(crate) fn set_payload_optional_content_type<B>(
```

Comment:
The original `set_payload` does two things: set the `content_type` and truncate the payload. In the PR, the content type is only set from `set_payload` but never directly. Splitting both actions makes the `Item`'s API usage easier to understand (vs wondering what a `None` is for). For example:

```rs
// envelope.rs

fn set_payload(...) {
  self.headers.content_type = content_type;
  self.set_truncated_payload(...);
}

fn set_truncated_payload<B>(&mut self, payload: B) where B: Into<Bytes> {
  // ...
}

// multipart.rs

if let Some(content_type) = field.content_type() {
  item.set_payload(content_type, field_data);
} else {
  item.set_truncated_payload(field_data);
}
```

(not opinionated on the function name.)

Extending the docstrings to point to the other function, depending on the `content_type` needs, would be great.